Added a feature to be able to select the appliance to connect by action

userlocalhost opened the pull request:
### Background
This pack has the following subjects.
* StackStorm can't load config.schema file because of the wrong suffix of it.
* The target appliance which is handled by each actions is static (as reported by #4).

### Action
These are main points of this patch.
* Renaming the file extension of config.schema file from 'yml' to 'yaml'.
* Adding a parameter 'appliance' to specify the appliance to connect for each actions.

userlocalhost committed:
Renamed the file extension of config.schema from 'yml' to 'yaml' to be able to load it from 'st2 pack config' command

userlocalhost committed:
Changed to be able to specify the target appliance to operate for each actions

userlocalhost committed:
Update of the tests accompanying the modification of adding parameter to
be able to specify the target appliance.

LindsayHill commented on actions/add_slb_server.yml:
```diff
@@ -37,3 +37,6 @@ parameters:
         default: False
+    appliance:
+        required: true
+        default: default
```
I think that this parameter should declare a type, for example: `type: string`.

I also think that it should not be required. I don't think it makes sense to have it `required: true` AND have a default value.

(Same comments apply for all the places you've added the `appliance` parameter

LindsayHill commented on config.schema.yml:
```diff
@@ -1,17 +0,0 @@
----
```
I don't know how I missed this the first time around. Sorry, I should have identified it when reviewing. 

There was some inconsistency in the past with what ST2 allowed, for `yml` vs `yaml`. But it should be all `.yaml` now.

userlocalhost replied:
> But it should be all .yaml now.

I got it. I'll change the suffix of all actions and configurations to `yaml`.

LindsayHill commented on actions/add_slb_service_group.yml:
```diff
@@ -57,3 +57,6 @@ parameters:
         default: round-robin
+    appliance:
+        required: true
+        default: default
```
I think that all of these action metadata files should be `.yaml`, not `.yml`

LindsayHill commented on config.schema.yaml:
```diff
@@ -0,0 +1,20 @@
+appliance:
+  type: object
+  properties:
+    default:
```
I don't _think_ this is right. I think it works if you have just one `default` appliance setting, but it won't work if you declare multiple appliance settings. I think it needs to look more like this: https://github.com/StackStorm-Exchange/stackstorm-mysql/blob/master/config.schema.yaml

userlocalhost replied:
Thank you for your advice @LindsayHill . I corrected it in [this commit](https://github.com/StackStorm-Exchange/stackstorm-acos/pull/6/commits/bbe76c98e556573ad3a2f01f9ab09e76c88685c7). But I'm not sure that is the best one.

I understood that I should have to write the config.schema to be able to specify multiple configuration-sets (`target IP address`, `username` and `password`) to connect. And I corrected `config.schema.yaml` following the one of MySQL pack.

But I have no confidence that the corrected `config.schema.yaml` is right.

In the `st2 config mysql` command, the properties which are specified in the `config.schema.file` (e.g. `host`, `user`, ...) aren't asked interactively. This is counterintuitive, I guess.

I'm not familiar with the implementation to parse the configuration schema file. So I'll make sure to know it.
If you know the best way to do it, could you please tell me that?

userlocalhost commented:
Thank you very much for your review and giving some important advices.
I'll fix them soon.

userlocalhost committed:
Renamed the suffix of all YAML actions and configurations based on the other StackStorm files

userlocalhost committed:
Changed to be able to specify mltiple configuration-sets to connect

userlocalhost committed:
Changed action configurations based on the StackStorm style

Concretely, changed to
- remove `required` attribute from the parameters which are specified
  `default` attribute.
- add `type` and `description` attribute at the `appliance` parameter.

userlocalhost committed:
Enabled to set multiple appliance to manage in the configuration file

userlocalhost commented:
In the last commit, I corrected this patch as following.

* Replaced the specification of API version parameter from action metadata to the appliance setting in the configuration file because it is related to the appliance.
* Allowed to set multiple setting of appliance information in the configuration file and User can specify it from action parameter.

userlocalhost committed:
Removed old-style default configuration value

 Previous implementation could specify only one appliance to operate by
the `appliance` parameter in configuration. So the default parameter was
specified in the config_schema.
 From this patch, multiple appliance can be specified. Therefore, the
action parameter of `appliance` will be mandatory in the
'GET/ADD/DELETE' action.

userlocalhost committed:
Filtered extended parameters that are not used by ACOS Client.

 Previous implementation passes arguments which is defined in the
acos-pack (e.g. 'target_path', ) to each actions of ACOS-Client.
 This may generate a harmful effect at the case of specifying the same
name parameters as the one used by ACOS-Client.
 In this commit, a processing to filter the extended paramters of
acos-pack is implemented.

userlocalhost commented:
I fixed remaining works below which were associated with the change of this PR.
- Removing old-style default configuration value (6411aa8)
- Filtering extended parameters that are not used by ACOS Client (cb8a249)

userlocalhost committed:
Corrected the explanation for the configuration in README based on the changing of #6

userlocalhost committed:
Added the description of change in this version

LindsayHill committed:
Small typo

LindsayHill reviewed (approved):
Looking good - ready to merge now?

userlocalhost commented:
Thank you for your review and fixing. Yes, it is!
